addons/addon-base-raas/packages/base-raas-workflow-steps/lib/steps/__test__/provision-account.test.js

```diff
@@ -0,0 +1,406 @@
+
+  const meta = { workflowId: `wf-provision-account` };
+  const input = {
+    requestContext,
+  };
+
+  const reporter = jest.fn();
+  reporter.print = jest.fn();
+
+  let container;
+  let step;
+
+  beforeAll(async () => {
+    container = new ServicesContainer();
+    const settingsService = new SettingsService();
+    settingsService.get = jest.fn(key => {
+      if (key === 'customUserAgent') {
+        return 'AwsLabs/SO0144/X.Y.Z';
+      }
+      throw new Error('Unexpected key');
+    });
+
+    container.register('settings', settingsService);
+    container.register('aws', new AwsService());
+    await container.initServices();
+  });
+  beforeEach(async () => {
+    step = new ProvisionAccount({
+      step: { config: {} },
+      stepReporter: reporter,
+      workflowPayload: new WorkflowPayload({
+        meta,
+        input,
+        workflowInstance: { steps: ['st-provision-account'] },
+      }),
+    });
+
+    step.payload = {
+      string: stringInput => {
+        return stringInput;
+      },
+      object: () => {
+        return requestContext;
+      },
+    };
+  });
+
+  describe('saveAccountToDb', () => {
+    beforeEach(() => {
+      const accountService = {
+        saveAccountToDb: jest.fn(),
+      };
+      accountService.saveAccountToDb = jest.fn();
+      step.mustFindServices = jest.fn().mockImplementation(async services => {
+        if (services[0] === 'accountService') {
+          return Promise.resolve([accountService]);
+        }
+        return null;
+      });
+
+      const describeAccount = jest.fn().mockReturnValue({
+        promise: jest.fn().mockReturnValue({
+          Account: {
+            Arn: 'abc123',
+          },
+        }),
+      });
+      const organization = {
+        describeAccount,
+      };
+      step.getOrganizationService = jest.fn().mockReturnValue(organization);
+
+      step.state = {
+        setKey: jest.fn(),
+        ...step.payload,
+      };
+    });
+
+    it('AppStream: false', async () => {
+      step.settings = {
+        get: key => {
+          if (key === 'isAppStreamEnabled') {
+            return 'false';
+          }
+          throw new Error('Unexpected key');
+        },
+      };
+      const response = await step.saveAccountToDb();
+      expect(response).toMatchObject({
+        waitDecision: {
+          seconds: 300,
+          thenCall: { methodName: 'deployStack', params: '[]' },
+          type: 'wait',
+        },
+      });
+    });
+
+    it('AppStream: true', async () => {
+      step.settings = {
+        get: key => {
+          if (key === 'isAppStreamEnabled') {
+            return 'true';
+          }
+          throw new Error('Unexpected key');
+        },
+      };
+      const response = await step.saveAccountToDb();
+      expect(response).toMatchObject({
+        waitDecision: {
+          seconds: 300,
+          thenCall: { methodName: 'shareImageWithMemberAccount', params: '[]' },
```

Comment:
Add test for `shareImageWithMemberAccount` and `createAppStreamRoles ` methods as well?

Reply:
added